cli/flox/src/commands/environment.rs

```diff
@@ -1238,12 +1238,22 @@ impl Upgrade {
         let mut environment = concrete_environment.into_dyn_environment();
 
         let upgraded = environment
-            .upgrade(&flox, self.groups_or_iids)
+            .upgrade(&flox, &self.groups_or_iids)
             .context("upgrading environment failed")?
             .0;
 
-        for package in upgraded {
-            info!("⬆️  upgraded '{package}' in environment {description}");
+        if upgraded.is_empty() {
+            if self.groups_or_iids.is_empty() {
+                info!("ℹ️  All packages are already upgraded in environment {description}.");
```

Comment:
all packages are upgraded sounds kinda weird..

how about

```suggestion
                info!("ℹ️  All packages are already up-to-date in environment {description}.");
```

or Or "No package is out of date/needs to be upgraded",

Reply:
Using the "No packages need to be upgraded," I do think that's less awkward. I was trying to avoid using the word updated, because you could have a very old package that didn't need an upgrade, but that did need an update and then an upgrade

Reply:
yeah i just thought that our messaging guidelines said something about being positive, i.e. it is better to say "everything is great" than "nothing is bad".
No need to shed the bike on this